Sources/Knit/Container.swift

```diff
@@ -72,3 +53,72 @@ extension Container {
     }
 
 }
+
+// **NOTE**: The only place this should be created is from the ModuleAssembler which
+// is responsible for creating Containers.
+// This should not be promoted from `internal` access level.
+internal final class ContainerManager {
+
+    // swinjectContainer is weak since the container will own this manager
+    private weak var swinjectContainer: Swinject.Container!
+
+    // ContainerManager from the parent of the SwinjectContainer
+    private let parent: ContainerManager?
```

Comment:
Alternately we could get this though the swinjectContainer's parent

Reply:
The Swinject.Container parent is private.